### PR TITLE
[TCL30-43] Distance calculations not quite right

### PR DESCRIPTION
**Pull request by alxmcr**
## Description

I've been noticing that the distance from center calculations seem off. For example, when I share my location, and tap a pin that is right next to me on the map, I get a distance that looks like, but I know it is within 5k of me:

distance

## Related Issue

[TCL30-43](https://the-collab-lab.atlassian.net/browse/TCL30-43)

## Acceptance Criteria

- Change `km` to `mts` (or `miles`).

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

image

### After

image

## Testing Steps / QA Criteria

1. Checkout `team-v2-fix-bugs` branch.
2. Visit `map` & `list`.
3. `Map`: Click the pins.
4. `List`: You can see locations.

### CODE REVIEW

File: src/helpers/helpersAPI.js [hunk 1]
```diff
@@ -17,4 +17,36 @@ const sortLocationsByDistance = (locations = []) => {
   });
 };
 
-export { getLocationsFromResponseAPI, sortLocationsByDistance };
+const MEASUREMENTS = {
+  METER: 'mts.',
+  KILOMETERS: 'kms.',
+  MILES: 'miles.',
+};
```
Comment (laurenmbeatty) [inline src/helpers/helpersAPI.js]:
```suggestion
  METER: 'm',
  KILOMETERS: 'km',
  MILES: 'miles,
};
```

File: src/helpers/helpersAPI.js [hunk 2]
```diff
@@ -17,4 +17,36 @@ const sortLocationsByDistance = (locations = []) => {
...
+
+const convertDistanceMts = (distanceInMts = 0, measurement = 'miles') => {
```
Comment (laurenmbeatty) [inline src/helpers/helpersAPI.js:26]: Was the issue that we were showing meters, but calling it kilometers?

File: src/helpers/helpersAPI.js [hunk 3]
```diff
@@ -17,4 +17,36 @@ const sortLocationsByDistance = (locations = []) => {
...
+    case MEASUREMENTS.KILOMETERS:
+      result = distanceInMts / 1000;
+      break;
+    case MEASUREMENTS.MILES:
+      result = distanceInMts * 0.000621371;
+      break;
+    default:
+      result = distanceInMts;
+      break;
+  }
```
Comment (laurenmbeatty) [inline src/helpers/helpersAPI.js:42]: This is good forward thinking! We could have the measurement in the dialog change depending on the users location! (Not for this ticket though)

**Review by laurenmbeatty — COMMENTED** [review]
Just a comment and a question.

File: src/helpers/helpersAPI.js [hunk 4]
```diff
@@ -17,4 +17,36 @@ const sortLocationsByDistance = (locations = []) => {
...
+
+  switch (measurement) {
+    case MEASUREMENTS.METERS:
+      result = distanceInMts;
+      break;
+    case MEASUREMENTS.KILOMETERS:
+      result = distanceInMts / 1000;
+      break;
+    case MEASUREMENTS.MILES:
+      result = distanceInMts * 0.000621371;
```
Comment (segdeha) [inline src/helpers/helpersAPI.js:37]: Silly imperial units 😅

File: src/helpers/helpersAPI.js (hunk 2 above)
Comment (segdeha) [reply src/helpers/helpersAPI.js:26]: Or was the issue that we were using units that were hard to reason about (e.g., 0.1km)?

**Review by segdeha — CHANGES_REQUESTED** [review]
I'm not sure this is solving the right problem. I just went to the app from the preview link and saw the dot representing my location moving as I zoomed. The before and after screenshots you provide, @alxmcr, show a change in the distance, not just a conversion to different units. This change may be correct, but I suggest fixing the other issue first to see if you get results that match more closely what you'd expect (i.e. things close to you are the right distance away).

my location moves as I zoom

File: src/helpers/helpersAPI.js (hunk 2 above)
Comment (laurenmbeatty) [reply src/helpers/helpersAPI.js:26]: What I noticed in prod, was it was showing that something was many kilometers away from me (like 300-1000km), but I knew it was within easy biking/walking distance. The information wasn't correct.

File: src/helpers/helpersAPI.js (hunk 3 above)
Comment (alxmcr) [reply src/helpers/helpersAPI.js:42]: Yes! I just created the ticket [TCL30-62](https://the-collab-lab.atlassian.net/browse/TCL30-62) for it ^.

File: src/helpers/helpersAPI.js (hunk 2 above)
Comment (alxmcr) [reply src/helpers/helpersAPI.js:26]: Yep! 

- 'Main' problem identified was that we forgot to convert `meters` to `km` then incorrect information.
- Another issue was distance values it was less than one so small. It's a reason why now it's as meters (`toFixed(2)`).

This PR is trying to solve ^^.

File: src/helpers/helpersAPI.js (hunk 1 above)
Comment (alxmcr) [reply src/helpers/helpersAPI.js]: Thanks, I just changed those values 😄 .

**Comment by alxmcr** [comment]
> I'm not sure this is solving the right problem. I just went to the app from the preview link and saw the dot representing my location moving as I zoomed. The before and after screenshots you provide, @alxmcr, show a change in the distance, not just a conversion to different units. This change may be correct, but I suggest fixing the other issue first to see if you get results that match more closely what you'd expect (i.e. things close to you are the right distance away).
> 
> my location moves as I zoom

@segdeha: Thanks for reporting this new bug.

However, I think it's not part of this PR. But, I suggest that we can create a new ticket bug at Jira: "Maps pins shouldn't move when you zoom in". Maybe I can be wrong.

Why?
- Because, I read a little about this bug, and it's more related to Google Markers (or pins our case - "User pin"). We should add new props to pins like `icon`, `anchor`, `zIndex`, and so on. I'm not sure but map props too.

What do you think?

File: src/helpers/helpersAPI.js (hunk 1 above)
Comment (alxmcr) [reply src/helpers/helpersAPI.js]: 😄  Good idea, @laurenmbeatty!

**Comment by segdeha** [comment]
> @segdeha: Thanks for reporting this new bug.
> 
> However, I think it's not part of this PR. But, I suggest that we can create a new ticket bug at Jira: "Maps pins shouldn't move when you zoom in". Maybe I can be wrong.
> 
> Why?
> 
> * Because, I read a little about this bug, and it's more related to Google Markers (or pins our case - "User pin"). We should add new props to pins like `icon`, `anchor`, `zIndex`, and so on. I'm not sure but map props too.
> 
> What do you think?

Yeah, I think you're right! What I noticed when I tried it again is that new pins are not fetched when I zoom in and out, which tells me that the map center is not changing when the zoom level changes. I think that means the center dot moving is just a display issue?

**Comment by alxmcr** [comment]
> > @segdeha: Thanks for reporting this new bug.
> > However, I think it's not part of this PR. But, I suggest that we can create a new ticket bug at Jira: "Maps pins shouldn't move when you zoom in". Maybe I can be wrong.
> > Why?
> > 
> > * Because, I read a little about this bug, and it's more related to Google Markers (or pins our case - "User pin"). We should add new props to pins like `icon`, `anchor`, `zIndex`, and so on. I'm not sure but map props too.
> > 
> > What do you think?
> 
> Yeah, I think you're right! What I noticed when I tried it again is that new pins are not fetched when I zoom in and out, which tells me that the map center is not changing when the zoom level changes. I think that means the center dot moving is just a display issue?

I just created a new [ticket](https://the-collab-lab.atlassian.net/browse/TCL30-63) Can we approval this PR?

**Comment by alxmcr** [comment]
> > > @segdeha: Thanks for reporting this new bug.
> > > However, I think it's not part of this PR. But, I suggest that we can create a new ticket bug at Jira: "Maps pins shouldn't move when you zoom in". Maybe I can be wrong.
> > > Why?
> > > 
> > > * Because, I read a little about this bug, and it's more related to Google Markers (or pins our case - "User pin"). We should add new props to pins like `icon`, `anchor`, `zIndex`, and so on. I'm not sure but map props too.
> > > 
> > > What do you think?
> > 
> > 
> > Yeah, I think you're right! What I noticed when I tried it again is that new pins are not fetched when I zoom in and out, which tells me that the map center is not changing when the zoom level changes. I think that means the center dot moving is just a display issue?
> 
> I just created a new [ticket](https://the-collab-lab.atlassian.net/browse/TCL30-63) Can we approval this PR?

@segdeha  What do you think about it^^?

**Comment by alxmcr** [comment]
I just reviewed it about zoom changes with this method:

```js
map.addListener("zoom_changed", () => {
      console.log(map)
      const center = map.getCenter()
      console.log(center.lat())
      console.log(center.lng())
});
```

I could confirm that our map center doesn't change. For me, this makes sense. However, I understood why the distance values change because you move the map, the center changes too 😄 .

image

If there isn't a contrary idea, I believe that we can merge this PR, and work with pins later. @segdeha .

**Review by segdeha — APPROVED** [review]
This looks good! Nice work!

**Pull request merged**